typed/symdiff_test.go

```diff
@@ -0,0 +1,483 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package typed
+
+import (
+	"fmt"
+	"testing"
+
+	"sigs.k8s.io/structured-merge-diff/fieldpath"
+	"sigs.k8s.io/structured-merge-diff/schema"
+	"sigs.k8s.io/structured-merge-diff/value"
+
+	"gopkg.in/yaml.v2"
+)
+
+type symdiffTestCase struct {
+	name         string
+	rootTypeName string
+	schema       string
+	quads        []symdiffQuad
+}
+
+type symdiffQuad struct {
+	lhs      string
+	rhs      string
+	modified *fieldpath.Set
+	added    *fieldpath.Set
```

Comment:
I understand that you're checking in both direction, but the tests are sort of hard to read or validate since you're not tracking "removed".

Reply:
added.